### ISSUE TITLE
lingui extract format --minimal not converting old messages.json files and produces faulty statistics.

### ISSUE THREAD

**Issue by np-8** [lingui/js-lingui#140]
The [lingui extract](https://lingui.github.io/js-lingui/ref/lingui-cli.html#extract) --format works when converting existing messages.json files from [minimal](https://lingui.github.io/js-lingui/ref/lingui-conf.html#minimal) to [lingui](https://lingui.github.io/js-lingui/ref/lingui-conf.html#lingui), but not vice versa. The output format is wrong, and the statistics are faulty (claims that no translations are missing).

## Setup
lingui-cli@1.4.2, lingui-react@1.4.1, babel-plugin-lingui-extract-messages@1.1.10, lingui-conf@0.12.0, Windows 10

## How to reproduce?

Using following test component:
```
import React from 'react'
import { Trans, Plural } from 'lingui-react'

const Testi18n = () => {
  return (
    <div>
      <Trans>This is a test</Trans>
      <Plural
        value={messagesCount}
        one="There's # message in your inbox"
        other="There're # messages in your inbox"
      />
    </div>
  )
}

export default Testi18n
```

(1) First, add locales -> will have messages.json files with only "{}" as content.

(2) `lingui extract --format minimal` produces messages.json in the following format:

```
{
  "This is a test": "",
  "{messagesCount, plural, one {There's # message in your inbox} other {There're # messages in your inbox}}": ""
}
```
and following statistics
```
Catalog statistics:
┌──────────┬─────────────┬─────────┐
│ Language │ Total count │ Missing │
├──────────┼─────────────┼─────────┤
│ en_US    │      2      │    2    │
│ fi_FI    │      2      │    2    │
└──────────┴─────────────┴─────────┘
```
(3) `lingui extract --format lingui` produces messages.json in following format:

```
{
  "This is a test": {
    "origin": [
      [
        "reactjs\\scenes\\Testi18n.jsx",
        7
      ]
    ]
  },
  "{messagesCount, plural, one {There's # message in your inbox} other {There're # messages in your inbox}}": {
    "origin": [
      [
        "reactjs\\scenes\\Testi18n.jsx",
        8
      ]
    ]
  }
}
```
and following statistics
```
Catalog statistics:
┌──────────┬─────────────┬─────────┐
│ Language │ Total count │ Missing │
├──────────┼─────────────┼─────────┤
│ en_US    │      2      │    2    │
│ fi_FI    │      2      │    2    │
└──────────┴─────────────┴─────────┘
```

(4) Running `lingui extract --format minimal` produces messages.json files in this format:
```
{
  "This is a test": {
    "origin": [
      [
        "reactjs\\scenes\\Testi18n.jsx",
        7
      ]
    ]
  },
  "{messagesCount, plural, one {There's # message in your inbox} other {There're # messages in your inbox}}": {
    "origin": [
      [
        "reactjs\\scenes\\Testi18n.jsx",
        8
      ]
    ]
  }
}
```
and following statistics
```
Catalog statistics:
┌──────────┬─────────────┬─────────┐
│ Language │ Total count │ Missing │
├──────────┼─────────────┼─────────┤
│ en_US    │      2      │    0    │
│ fi_FI    │      2      │    0    │
└──────────┴─────────────┴─────────┘
```

**Comment by tricoder42** [comment]
I never intended to support this. Conversion from minimal to lingui is for backward compatibility.

I think we could either add `convert` command or `--from` option to `lingui extract` command to make conversion explicit.

**Comment by np-8** [comment]
In that case I think, that `lingui extract` should just throw an error if the format is not matching. 

Then, conversion (from `minimal` too `lingui`) should be done by something like  `lingui convert-format-to lingui`. If using `--from` -flags, then adding a new format some day could cause problems (how would you know to which format you want to convert if you have multiple options?). Also, since the conversion is not reversible, there should be big warnings and it should ask for user input ("`This conversion is not reversible. Do you want to proceed?`"). 

I also think that `minimal` would better suite to be the default for these reasons
1) `lingui` does not anything to the table.  (referencing to [this](https://github.com/lingui/js-lingui/issues/138#issuecomment-352269065))
2) Some translation tools really like the `minimal` JSON format (and are not happy with `lingui` format)
3) Conversion from `minimal` to `lingui` is possible. The other direction is not. This way you can change your mind later.

**Comment by tricoder42** [comment]
Yes, I agree `minimal` format should be default. Unfortunatelly it's backward incompatible change, so it'll land in next major release.

**Comment by tricoder42** [comment]
> If using --from -flags, then adding a new format some day could cause problems (how would you know to which format you want to convert if you have multiple options?)

Format is defined using `"format": "lingui"` config. Source format would be taken from `--from` option and target from config.

**Comment by np-8** [comment]
You're right, both, _from_ and _to_, must be defined somehow. Using the "to" option from the config makes sense, and therefore using a `--from` flag is natural.

**Comment by tricoder42** [comment]
Maybe `--from-format` or `--convert-format` would be better.

I've just added draft of [release docs](https://github.com/lingui/js-lingui/blob/master/docs/releases/migration-2.rst)

**Comment by np-8** [comment]
Nice! 👍  I think it would be even clearer to have explicit `lingui convert --from lingui` -command, instead of bundling it with the `lingui extract` -command. But that's just a matter of taste.

**Comment by mcku** [comment]
I think adding this feature would be useful. Because the extracted messages.json may get uploaded to 3rd party translation services. I don't know if any such services support lingui format.

**Comment by tricoder42** [comment]
I agree, `lingui` format is overkill for most usecases. The change is ready in `next-2` branch and will be released soon in [2.x release](https://github.com/lingui/js-lingui/blob/next-2/docs/releases/migration-2.rst#default-message-catalog-format).

**Comment by tricoder42** [comment]
Fixed in `@lingui/cli@2.0.0`. See [release docs](https://lingui.github.io/js-lingui/releases/migration-2.html#default-message-catalog-format)
